app/containers/Register/index.js

```diff
@@ -13,82 +13,177 @@ import {
     Button, 
     Text
 } from 'native-base';
-import { StyleSheet } from 'react-native';
+import { StyleSheet, Alert } from 'react-native';
+import ValidationComponent from 'react-native-form-validator';
+
 const role = [
-        { 
-            value: "key0",
-            label: "Attendee"
-        },
-        { 
-            value: "key1",
-            label: "Booth"
-        },
-        { 
-            value: "key2",
-            label: "Speaker"
-        },
-        { 
-            value: "key3",
-            label: "Admin"
-        },
-    ]
+    { 
+        value: "key0",
+        label: "Attendee"
+    },
+    { 
+        value: "key1",
+        label: "Booth"
+    },
+    { 
+        value: "key2",
+        label: "Speaker"
+    }
+];
 
-export default class Register extends Component {
+export default class Register extends ValidationComponent {
     constructor(props) {
         super(props);
         this.state = {
-            selected1: 'key0'
+            firstName: '',
+            lastName: '',
+            role: '',
+            email: '',
+            password: '',
+            username: '',
+            isError: {
+                "firstName": false,
+                "lastName": false,
+                "email": false,
+                "password": false,
+                "username": false,
+            }
         }
     }
+
     onValueChange (value: string) {
         this.setState({
-            selected1 : value
+            role : value
+        });
+    }
+
+    componentWillMount(){
+        this.isFieldInError('firstName')
+        this.setState({role: 'Attendee'});
+    }
+
+    handleInputChange = async (field, value) => {
+        await this.setValue(field, value)
+
+        const rules = {
```

Comment:
you can move this outside of the class as constants, so you don't write it twice